standalone/pherry/src/lib.rs

```diff
@@ -249,34 +257,49 @@ async fn get_header_at<T: subxt::Config>(
     Ok((header, hash))
 }
 
-async fn get_block_without_storage_changes(
-    api: &RelaychainApi,
-    h: Option<u32>,
-) -> Result<BlockWithChanges> {
+async fn get_block_without_storage_changes(api: &RelaychainApi, h: Option<u32>) -> Result<Block> {
     let (block, hash) = get_block_at(&api.client, h).await?;
     info!("get_block: Got block {:?} hash {}", h, hash.to_string());
-    return Ok(BlockWithChanges {
-        block,
-        storage_changes: Default::default(),
-    });
+    return Ok(block);
 }
 
-pub async fn get_block_with_storage_changes(
+pub async fn batch_get_storage_changes(
     api: &ParachainApi,
-    h: Option<u32>,
-) -> Result<BlockWithChanges> {
-    let (block, hash) = get_block_at(&api.client, h).await?;
-    info!(
-        "get_block (w/changes): Got block {:?} hash {}",
-        h,
-        hash.to_string()
-    );
-    let hash = block.block.header.hash();
-    let storage_changes = chain_client::fetch_storage_changes(api, &hash).await?;
-    return Ok(BlockWithChanges {
-        block,
-        storage_changes,
-    });
+    from: BlockNumber,
+    to: BlockNumber,
+    batch_size: BlockNumber,
+) -> Result<Vec<BlockHeaderWithChanges>> {
+    info!("batch_get_changes from {from} to {to} ({} blocks)", to as i64 - from as i64 + 1);
+
+    let mut changes = vec![];
+
+    for from in (from..=to).step_by(batch_size as _) {
+        let to = to.min(from + batch_size - 1);
+        let from_hash = get_header_hash(&api.client, Some(from)).await?;
+        let to_hash = get_header_hash(&api.client, Some(to)).await?;
+        info!("batch_get_changes one batch from {from:?} to {to:?}");
+        let storage_changes =
+            chain_client::fetch_storage_changes(api, &from_hash, &to_hash).await?;
+        changes.extend(
+            storage_changes
+                .into_iter()
+                .enumerate()
+                .map(|(offset, storage_changes)| {
+                    BlockHeaderWithChanges {
+                        // Only the block number is used in pRuntime.
```

Comment:
Suggest to change it to "Headers are synced separately. Only the block number is used in block sync."

Reply:
Done